tests/executables_src/testLMapMathPluginPushPars.cc

```diff
@@ -47,73 +48,120 @@ BOOST_AUTO_TEST_CASE(testPushPars) {
     // Test needs to call activateAsyncRead but it must be irrelevant whether before or after writes
 
     ChimeraTK::Device targetDevice;
-    targetDevice.open("HOLD");
+    auto targetWriteCount = [&targetDevice]() {
+      auto exceptionDummyForTargetDev = boost::static_pointer_cast<ExceptionDummy>(targetDevice.getBackend());
+      return exceptionDummyForTargetDev->getWriteCount("MATHTEST/TARGET");
+    };
+    size_t writeCount = 0; // this counter tracks expected writes to target register
 
-    auto accTarget = targetDevice.getScalarRegisterAccessor<uint32_t>("HOLD0/WORD_G");
-    auto pollPar = targetDevice.getScalarRegisterAccessor<uint32_t>("HOLD0/POLLPAR");
-    pollPar = 1;
-    pollPar.write();
+    targetDevice.open("HOLD");
+    auto accTarget = targetDevice.getScalarRegisterAccessor<uint32_t>("MATHTEST/TARGET");
 
-    ChimeraTK::Device device;
-    device.open("EOD");
-    device.activateAsyncRead();
-    auto pushPar = device.getScalarRegisterAccessor<uint32_t>("DET/EXPOSURE");
+    ChimeraTK::Device logicalDevice("EOD");
+    logicalDevice.open();
+    logicalDevice.activateAsyncRead();
+    auto pushPar = logicalDevice.getScalarRegisterAccessor<uint32_t>("DET/PUSHPAR");
 
     pushPar = 2;
     pushPar.write();
 
-    auto accMathWrite = device.getScalarRegisterAccessor<double>("DET/GAIN");
+    auto accMathWrite = logicalDevice.getScalarRegisterAccessor<double>("DET/X");
```

Comment:
:+1:  X is a very good name for the main variable because it's the same as in the formula